### PR TITLE
fix(pagination)!: gate PaginatedQueryRet construction and take

### PR DESCRIPTION
**Pull request by vindard**
Follow-up to #228, which introduced a public `page_size` field on `PaginatedQueryRet` as a breaking change. This explores gating construction and making entity extraction a single atomic operation, so future internal field changes stop being breaking.

Stack (all on top of the #228 branch):
1. **Gate construction** — `PaginatedQueryRet::new(entities, has_next_page, end_cursor, requested_size)`; `requested_size` is private and construction-derived. Generated repos construct via `new` and convert cursors through `map_end_cursor` instead of destructuring private fields.

2. **Gate `entities`** — the vec is private; read it via `entities()`, move it out via a dedicated method.

3. **Take** — `into_parts(self) -> (Vec<T>, Option<PaginatedQueryArgs<C>>)` consumes the page and returns the entities together with the precomputed next query in one step. Draining is atomic; there is no residual page object left to misuse.

### Design rationale

Pagination callers need ownership of the entities to accumulate across pages, so forbidding the take would force per-page clones. The take is therefore facilitated, but atomically — a single consuming method, no dedicated continuation type — giving the post-drain safety guarantee by consumption (`compile_fail` doctest) rather than by machinery.

---

> [!NOTE]
> **Medium Risk**
> Breaking public pagination API across generated repos and downstream callers; behavior change fixes continuation page size but requires migration from `.entities` and direct struct construction.
> 
> **Overview**
> **Breaking change:** `PaginatedQueryRet` no longer exposes a public `entities` field or allows struct-literal construction. Repos and macros now build pages with **`PaginatedQueryRet::new(..., requested_size)`**, where **`requested_size`** is the original `first` (kept private) so continuation queries no longer incorrectly use **`entities.len()`** on short pages.
> 
> Callers read rows via **`entities()`** and collect across pages with **`into_parts()`** — entities plus an optional next **`PaginatedQueryArgs`** in one consuming step. **`into_next_query()`** delegates to that logic; **`map_end_cursor`**, **`map_entities`**, and **`try_map_entities`** preserve pagination metadata. Generated **`list_for_filters`** dispatch uses **`map_end_cursor`** instead of manual destructuring.
> 
> Docs and tests are updated to the new API, with unit and integration tests asserting stable page size when draining pages (including default `first: 100` and **`first: 0`** edge cases).
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit f849cb5bdf4dfce75dd78eb61203909c6fa2c42d. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: book/src/repo-list-for-filters.md [hunk 1]
```diff
@@ -207,7 +207,7 @@ async fn main() -> anyhow::Result<()> {
     ).await?;
 
     // Paginate through results
-    if let Some(next_query) = filtered.into_next_query() {
+    if let Some(next_query) = filtered.into_parts().1 {
```
Comment (bodymindarts) [inline book/src/repo-list-for-filters.md:210]: This api change looks strictly worse.

Comment (vindard) [reply book/src/repo-list-for-filters.md]: > This api change looks strictly worse.

Yup we can iterate on this. The current state I'm trying to avoid is to drain the entities but still have a mutated usuable type (current state in David's PR). I also want to keep the behavior that whatever returns `next` makes the original object unusable (takes) to signal that we have moved on.

Maybe having a parallel function that just returns the next type as an alternative is better semantically (I'll try that). 

_**Edit:** a rework here https://github.com/GaloyMoney/es-entity/pull/229/commits/64e4664bddd6335a19aca81ddbf406eeec840991_

**Comment by vindard** [comment]
Concrete Lana usage supports keeping all three helpers: four adapter sites need cursor mapping, while entity mapping has one infallible and three fallible uses. The full call-site breakdown and necessity assessment are below.

<details>
<summary>Full Lana call-site and abstraction analysis</summary>

These methods are not in Lana’s pinned `es-entity 0.12.20`; they’re on the local `es-entity` pagination branch. After upgrading, there are **four concrete adapter sites**.

| Helper | Current Lana uses |
|---|---:|
| `map_end_cursor` | 4 |
| `map_entities` | 1 |
| `try_map_entities` | 3 |

### 1. Infallible entity and cursor conversion

`core/deposit/src/ledger/mod.rs:367`

Current:

```rust
let entities = ret.entities.into_iter().map(T::from).collect();

Ok(es_entity::PaginatedQueryRet {
    entities,
    has_next_page: ret.has_next_page,
    end_cursor: ret.end_cursor.map(U::from),
})
```

Becomes:

```rust
Ok(ret
    .map_entities(T::from)
    .map_end_cursor(U::from))
```

This is the sole direct `map_entities` use.

### 2. Fallible entity and cursor conversion

`core/accounting/src/ledger_account/ledger/mod.rs:90`

```rust
Ok(ret
    .try_map_entities(T::try_from)?
    .map_end_cursor(U::from))
```

`core/accounting/src/journal/mod.rs:89`

```rust
Ok(ret
    .try_map_entities(LedgerEntry::try_from)?
    .map_end_cursor(LedgerEntryCursor::from))
```

### 3. Fallible enriched entity conversion

`core/accounting/src/ledger_transaction/mod.rs:155`

This one first bulk-loads each transaction’s entries, then converts every page entity:

```rust
let entry_ids = transactions
    .entities()
    .iter()
    .flat_map(|tx| tx.values().entry_ids.iter().copied())
    .collect::<Vec<_>>();

let mut all_entries = self.cala.entries().find_all(&entry_ids).await?;

let transactions = transactions.try_map_entities(|tx| {
    let mut entries = tx
        .values()
        .entry_ids
        .iter()
        .filter_map(|id| all_entries.remove(id))
        .collect::<Vec<_>>();

    entries.sort_by_key(|entry| entry.values().sequence);
    LedgerTransaction::try_from((tx, entries))
})?;

Ok(transactions.map_end_cursor(LedgerTransactionCursor::from))
```

So this uses `try_map_entities` despite the conversion being more substantial than `T::try_from`.

### Not a use: transaction templates

`core/accounting/src/transaction_templates/mod.rs:75` currently reconstructs an identical page:

```rust
Ok(es_entity::PaginatedQueryRet {
    entities: cursor.entities,
    has_next_page: cursor.has_next_page,
    end_cursor: cursor.end_cursor,
})
```

It should simply become:

```rust
Ok(cursor)
```

The GraphQL `.entities.into_iter().map(...)` sites are also **not** candidates: they are constructing GraphQL connections or plain vectors, not republishing `PaginatedQueryRet`.

## Are they necessary?

**Given the blueprint’s private page state: yes.**

Making `requested_size` and `entities` private establishes an invariant: adapters must not accidentally lose or recreate pagination metadata. Without these methods, callers cannot transform a page’s entity or cursor type while preserving:

- requested page size,
- `has_next_page`,
- final `end_cursor`, including on the last page.

`into_parts()` is not a substitute because it returns a continuation, not the raw cursor metadata; on a terminal page there is no continuation even though the page may still have an end cursor.

Individually:

- **`map_end_cursor` is clearly necessary.** Four Lana sites need it, and generated `es-entity` proxy methods use it internally.
- **`try_map_entities` is clearly necessary.** Three Lana domain adapters perform fallible conversions.
- **`map_entities` has only one current Lana call site**, but it is the canonical infallible counterpart. Requiring callers to express infallible mapping through `try_map_entities` and `Infallible` would be worse API design.

I would keep exactly these three. They are small orthogonal operations and compose cleanly:

```rust
page
    .try_map_entities(Entity::try_from)?
    .map_end_cursor(Cursor::from)
```

I would **not** add combined `map_page`, `try_map_page`, or `try_map_end_cursor` abstractions yet. There are no current Lana use cases earning those.

</details>

**Review by bodymindarts — APPROVED** [review]
